src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -455,10 +455,11 @@ private function createTypeNotFoundMessage(TypedReference $reference, $label)
 
             $message = sprintf('has type "%s" but this class %s.', $type, $parentMsg ? sprintf('is missing a parent class (%s)', $parentMsg) : 'was not found');
         } else {
+            $alternatives = $this->createTypeAlternatives($reference);
             $message = $this->container->has($type) ? 'this service is abstract' : 'no such service exists';
             $message = sprintf('references %s "%s" but %s.%s', $r->isInterface() ? 'interface' : 'class', $type, $message, $this->createTypeAlternatives($reference));
```

Comment:
$alternatives to be used on this line, no need to call createTypeAlternatives twice

Reply:
fixed